warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -533,6 +533,7 @@ private void configureIterator(ShardQueryConfiguration config, IteratorSetting c
         setCommonIteratorOptions(config, cfg);
         
         addOption(cfg, QueryOptions.LIMIT_FIELDS, config.getLimitFieldsAsString(), true);
+        addOption(cfg, QueryOptions.MATCHING_FIELD_SETS, config.getMatchingFieldSetsAsString(), true);
```

Comment:
The last argument should be false. Otherwise we will serialize the key with an empty value for every range generated. In fact...shouldn't most of these options have flag set to false?

Reply:
ah, ok.